### PR TITLE
espeak: new port

### PR DESCRIPTION
**Pull request by danielbair**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.5 19F101
Xcode 11.5 11E608c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by reneeotten** [comment]
thanks for the submission @danielbair! 

I only looked quickly, but it seems to me that the port is not using [TheRightCompiler](https://trac.macports.org/wiki/UsingTheRightCompiler). From the logs you can see it uses `c++` whereas it should be using ${configure.cxx}, which defaults to `/usr/bin/clang++`. Perhaps using the `makefile` PG here would be useful to propagate MacPorts's settings in the absence of a configure phase.
Additionally, in your patch you are hard-coding the PREFIX as /opt/local; that's not necessarily correct - although I see that you also supply this as `build.args`/`destroot.args` again. So perhaps defining it in the Makefile is not needed?

File: audio/espeak/Portfile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    depends_lib-append      port:portaudio
+    configure.args-append   --with-mictype=portaudio
+}
+
+use_configure       no
+build.dir           ${worksrcpath}/src
+build.args          PREFIX=${prefix} DATADIR=${prefix}/share/espeak-data 
+build.args-append   CC=${configure.cc} \
+                    CXX=${configure.cxx} \
+                    CPP=${configure.cpp}
```
Comment (reneeotten) [inline audio/espeak/Portfile:40]: these two `build.args[-append]` statements can be combined into one

**Comment by reneeotten** [comment]
thanks @danielbair

**Comment by ryandesign** [comment]
@danielbair Thanks for the port. There are many problems with the port that I'm working on fixing, but where did the enormous patchfile come from? That looks like it will be impossible for us to maintain.

**Comment by danielbair** [comment]
> where did the enormous patchfile come from? That looks like it will be impossible for us to maintain.

The install instructions from the developer state that portaudio19.h should be copied to portaudio.h BUT if there is a pre-build stage that can do that, it would make the patch file much smaller. 

I also just learned that the correct flag for creating a dynamic library (Apple's name for a shared library) is -dynamiclib not -shared. (line 117 in the Makefile)

**Comment by ryandesign** [comment]
> The install instructions from the developer state that portaudio19.h should be copied to portaudio.h BUT if there is a pre-build stage that can do that, it would make the patch file much smaller.

Ok great I'll change it do that and then the patch will be much smaller.

> I also just learned that the correct flag for creating a dynamic library (Apple's name for a shared library) is -dynamiclib not -shared. (line 117 in the Makefile)

Right, that's one of the changes I already made in d9b027b58574a9a9b6a5f204153144ef8a58aa9f.

**Pull request merged**